projects/python-client/src/datapane/client/config.py

```diff
@@ -3,35 +3,38 @@
 We use a module-level singleton pattern here, where top-level functions and globals
 act like the state for a "Manager" class around the internal Config object
 """
+import configparser
 import dataclasses as dc
 import typing as t
 import uuid
 import webbrowser
 from contextlib import suppress
+from os import getenv
 from pathlib import Path
 from typing import Optional
 
 import click
-import dacite
-import yaml
 from furl import furl
 
 from datapane import _IN_PYTEST, log
 
 from .utils import InvalidTokenError
 
 APP_NAME = "datapane"
-APP_DIR = Path(click.get_app_dir(APP_NAME))
+APP_DIR = Path(getenv("DATAPANE_APP_DIR", click.get_app_dir(APP_NAME)))
 APP_DIR.mkdir(parents=True, exist_ok=True)
-DEFAULT_ENV = "default"
+CONFIG_FILENAME = "config.ini"
+LEGACY_CONFIG_FILENAME = "default.yaml"
+CONFIG_PATH = Path(getenv("DATAPANE_CONFIG", APP_DIR / CONFIG_FILENAME))
+LEGACY_CONFIG_PATH = APP_DIR / LEGACY_CONFIG_FILENAME
+CONFIG_SECTION = "default"
 PAST_DEFAULT_SERVER = "https://datapane.com"
 DEFAULT_SERVER = "https://cloud.datapane.com"
 DEFAULT_TOKEN = "TOKEN_HERE"
-LATEST_VERSION = 4
+LATEST_VERSION = 6
 
 
 # TODO - wrap into a singleton object that includes callable?
-# TODO - switch to pydantic
 @dc.dataclass
 class Config:
```

Comment:
Happy for us to drop the `Config` dataclass and just use the `configparser` object directly (or wrapped in a helper object) if that makes loading/saving easier?

Reply:
Had another look at the options; configparser doesn't do any typing, so I think keeping it as a dataclass is still the easiest option, at least for now. We might be able to simplify it further if we moved to toml, but that would mean introducing another dependency and it seems like little benefit at this stage.

Reply:
toml is moving into the standard library in 3.11 I think - but will be a while before we support that as a minimum!

Happy to keep as-is then - to go further, I think it's ok for us to drop typing as part of this in the name of simplicity. But can look at in a future cooldown